src/Generators/VsixManifestGenerator.cs

```diff
@@ -9,48 +8,93 @@ namespace VsixSynchronizer
+        /// found in the specified <paramref name="manifest" />.
+        /// </summary>
+        /// <param name="manifest">
+        /// (Required.) Reference to an instance of
+        /// <see cref="T:VsixSynchronizer.VsixManifest" /> that provides the information to
+        /// be written to the generated code.
+        /// </param>
+        /// <returns>
+        /// String containing the text to be written to the output file.
+        /// <para />
+        /// If the <paramref name="manifest" /> parameter is passed a
+        /// <see langword="null" /> reference, then this method returns the empty string.
+        /// </returns>
         private string GenerateClass(VsixManifest manifest)
         {
+            if (manifest == null)
+                return string.Empty;
+
             var sb = new StringBuilder();
-            sb.AppendLine($"// ------------------------------------------------------------------------------");
-            sb.AppendLine($"// <auto-generated>");
-            sb.AppendLine($"//     This file was generated by {Vsix.Name}");
-            sb.AppendLine($"// </auto-generated>");
-            sb.AppendLine($"// ------------------------------------------------------------------------------");
+            sb.AppendLine(
+                "// ------------------------------------------------------------------------------"
+            );
+            sb.AppendLine("// <auto-generated>");
+            sb.AppendLine(
+                $"//     This file was generated by {Vsix.Name}.  Changes made to the code in this "
+            );
+            sb.AppendLine(
+                "//      file will be lost the next time that the project is built."
+            );
+            sb.AppendLine("// </auto-generated>");
+            sb.AppendLine(
+                "// ------------------------------------------------------------------------------"
+            );
 
             sb.AppendLine($"namespace {FileNamespace}");
             sb.AppendLine("{");
-            sb.AppendLine($"    internal sealed partial class Vsix");
-            sb.AppendLine($"    {{");
-            sb.AppendLine($"        public const string Id = \"{EscapeString(manifest.ID)}\";");
-            sb.AppendLine($"        public const string Name = \"{EscapeString(manifest.Name)}\";");
-            sb.AppendLine($"        public const string Description = @\"{EscapeVerbatimString(manifest.Description)}\";");
-            sb.AppendLine($"        public const string Language = \"{EscapeString(manifest.Language)}\";");
-            sb.AppendLine($"        public const string Version = \"{EscapeString(manifest.Version)}\";");
-            sb.AppendLine($"        public const string Author = \"{EscapeString(manifest.Author)}\";");
-            sb.AppendLine($"        public const string Tags = \"{EscapeString(manifest.Tags)}\";");
-            sb.AppendLine($"    }}");
+            sb.AppendLine("    internal sealed partial class VsixAttributes");
```

Comment:
Please don't rename the class. It would break all the extensions currently using it. Also, the naming collision has never been reported as an issue before, so probably isn't a problem.

Reply:
> Please don't rename the class. It would break all the extensions currently using it. Also, the naming collision has never been reported as an issue before, so probably isn't a problem.

Will do. I'll close this PR.

Name changes of classes breaking other code is why I am a **_strong_** advocate of always having classes implement an interface and using interfaces as they types of method parameters, fields, and properties.

Of course, it goes without saying that you cannot make a `static` class that exposes constants implement an interface, but in general, this is a best practice.  It also maximizes the ability to use the code with DI containers/IoC.